dwave/optimization/src/nodes/binaryop.cpp

```diff
@@ -367,25 +367,40 @@ template <class BinaryOp>
 std::span<const ssize_t> BinaryOpNode<BinaryOp>::shape(const State& state) const {
     if (!this->dynamic()) return this->shape();
 
-    const ssize_t lhs_size = operands_[0]->size(state);
+    const auto [lhs, rhs] = operands_;
 
-    if (lhs_size == operands_[1]->size(state)) return operands_[0]->shape(state);
```

Comment:
To explain, this line meant that if say you has a `set(5)` broadcasting with a `0`, they would have the same size and we therefore would return the shape of the rhs, which in that example would be the shape of `0`, i.e. empty.